.github/workflows/cicd-main.yml

```diff
@@ -817,6 +817,72 @@ jobs:
       AFTER_SCRIPT: |
         rm -rf stt_test_res.json
 
+  # L2: Speech Transcription
+  L2_Speech_Transcription_Canary_Transcribe_Full_Manifest:
+    needs: [cicd-test-container-setup]
+    uses: ./.github/workflows/_test_template.yml
+    with:
+      RUNNER: self-hosted-azure
+      SCRIPT: |
+        python examples/asr/transcribe_speech.py \
+        dataset_manifest=/mnt/datadrive/TestData/asr/canary/dev-other-wav-10-canary-fields.json \
+        output_filename=preds.json \
+        batch_size=10 \
+        pretrained_name=nvidia/canary-1b \
+        num_workers=0 \
+        amp=false \
+        compute_dtype=bfloat16 \
+        matmul_precision=medium \
+        2>&1 | tee transcribe.log
+        actual_wer=$(grep -oP "(?<='wer': )[0-9.]*" transcribe.log)
+        (( $(echo "$actual_wer >= 0.031 && $actual_wer <= 0.032" | bc -l) )) && exit 0 || exit 1
+      AFTER_SCRIPT: |
+        rm -rf preds.json transcribe.log
+
+  L2_Speech_Transcription_Canary_Transcribe_With_Prompt:
+    needs: [cicd-test-container-setup]
+    uses: ./.github/workflows/_test_template.yml
+    with:
+      RUNNER: self-hosted-azure
+      SCRIPT: |
+        python examples/asr/transcribe_speech.py \
+        dataset_manifest=/mnt/datadrive/TestData/asr/canary/dev-other-wav-10.json \
+        output_filename=preds.json \
+        batch_size=10 \
+        pretrained_name=nvidia/canary-1b \
+        num_workers=0 \
+        amp=false \
+        compute_dtype=bfloat16 \
+        matmul_precision=medium \
+        +prompt.source_lang="en" \
+        +prompt.target_lang="en" \
+        +prompt.task="asr" \
+        +prompt.pnc="no" \
+        2>&1 | tee transcribe.log
+        actual_wer=$(grep -oP "(?<='wer': )[0-9.]*" transcribe.log)
+        (( $(echo "$actual_wer >= 0.031 && $actual_wer <= 0.032" | bc -l) )) && exit 0 || exit 1
```

Comment:
Same as above. Keep it 0.03 +- 0.02.

Reply:
This is too wide, that would allow lots of variations in WER (1-5% absolute). For fixed data and fixed model the result should be 100% deterministic; and even with some oversights, there is no reason the WER should vary more than within 0.1%. Let me make it 0.03 - 0.032 (the current output is 0.03107 or so).

Reply:
FWIW you were right (the CI yields 3.8% WER) and I broadened the range because I don't have the time to look into this right now. For the record I find this... scary to say the least that we can get such a wide WER variance depending on the environment.

Reply:
Ouch. Just between runs or between config changes ? Between same runs shouldn't have that wide a spread normally

Reply:
Between my workstation and the CI environment. It is stable in the CI though (doesn't vary) which gives hopes for debugging it.

BTW I removed the WER check altogether, turning this test into "it can run" for now similarly to other tests in the suite. I was getting bash errors that didn't make sense in the CI env and I'd rather merge this ASAP since multiple people are experiencing these bugs. I'll try to restore these checks later when CI is more stable, letting me experiment a bit.